openupgrade_scripts/scripts/fleet/18.0.0.1/post-migration.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2025 Open Net Sàrl
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+def _set_power_unit(env):
+    # if horspower is defined, set power_unit to horsepower
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE fleet_vehicle_model
+        SET power_unit='horsepower'
+        WHERE horsepower > 0
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE fleet_vehicle
+        SET power_unit='horsepower'
+        WHERE horsepower > 0
+        """,
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    _set_power_unit(env)
+    data = env["ir.model.data"].search(
+        [
+            ("module", "=", "fleet"),
+            (
+                "name",
+                "in",
+                [
+                    "model_a1",
+                    "model_a3",
+                    "model_a4",
+                    "model_a5",
+                    "model_a6",
+                    "model_a7",
+                    "model_a8",
+                    "model_agila",
+                    "model_ampera",
+                    "model_antara",
+                    "model_astra",
+                    "model_astragtc",
+                    "model_classa",
+                    "model_classb",
+                    "model_classc",
+                    "model_classcl",
+                    "model_classcls",
+                    "model_classe",
+                    "model_classgl",
+                    "model_classglk",
+                    "model_classm",
+                    "model_classr",
+                    "model_classs",
+                    "model_classslk",
+                    "model_classsls",
+                    "model_combotour",
+                    "model_corsa",
+                    "model_insignia",
+                    "model_meriva",
+                    "model_mokka",
+                    "model_q3",
+                    "model_q5",
+                    "model_q7",
+                    "model_serie1",
+                    "model_serie3",
+                    "model_serie5",
+                    "model_serie6",
+                    "model_serie7",
+                    "model_seriehybrid",
+                    "model_seriem",
+                    "model_seriex",
+                    "model_seriez4",
+                    "model_tt",
+                    "model_zafira",
+                    "model_zafiratourer",
+                ],
+            ),
+        ]
+    )
+    data.unlink()
```

Comment:
Why don't you use `delete_records_safely_by_xml_id` that tries to remove data if not used, or the XML-ID if not able to remove?

Reply:
Also side technical question : delete_records_safely_by_xml_id does not remove the XML-ID but change it to no-update if unable to remove. Why not delete the XML-ID ?

Reply:
> it was better to keep the models

In fact, Odoo didn't delete them. They moved them to demo data.

Reply:
> they are not there on non demo databases

Well, in v17, they were as non-demo data. So client coming from v17 with fleet installed, will have this data of vehicle models. And they could be used.

Reply:
If they are used, they can't be removed.

Reply:
Suppose that from those vehicle models, you used 3. Thus, following you rationale, the other vehicle models would be deleted except those 3. It doesn't make much sense. Or you have them all, or you remove them all.

Reply:
Well, I see it as the correct behavior